components/slideshow.tsx

```diff
@@ -0,0 +1,97 @@
+'use client';
+import Image from 'next/image';
+import { useCallback, useEffect, useState } from 'react';
+import { FaChevronLeft, FaChevronRight } from 'react-icons/fa';
+
+export default function Carousel({ images }: { images: string[] }) {
+  const [currentSlide, setCurrentSlide] = useState<number>(0);
+  const [hovered, setHovered] = useState<boolean>(false);
+
+  const prevSlide = useCallback(() => {
+    setCurrentSlide((prev: number) =>
+      prev === 0 ? images.length - 1 : prev - 1
+    );
+  }, [images.length]);
+
+  const nextSlide = useCallback(() => {
+    setCurrentSlide((prev: number) =>
+      prev === images.length - 1 ? 0 : prev + 1
+    );
+  }, [images.length]);
+
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      nextSlide();
+    }, 3000);
+
+    return () => clearInterval(intervalId);
+  }, [nextSlide]);
+
+  const handleKeyDown = useCallback(
+    (event: KeyboardEvent) => {
+      if (event.key === 'ArrowLeft') {
+        prevSlide();
+      } else if (event.key === 'ArrowRight') {
+        nextSlide();
+      }
+    },
+    [prevSlide, nextSlide]
+  );
+
+  useEffect(() => {
+    window.addEventListener('keydown', handleKeyDown);
+    return () => window.removeEventListener('keydown', handleKeyDown);
+  }, [handleKeyDown]);
+
+  return (
+    <div
+      className="relative overflow-hidden"
+      onMouseEnter={() => setHovered(true)}
+      onMouseLeave={() => setHovered(false)}
+    >
+      <button
+        className={`prev absolute top-1/2 -translate-y-1/2 transform ${
+          hovered ? 'opacity-100' : 'opacity-0'
+        } z-10 transition-opacity duration-300`}
+        onClick={prevSlide}
+        style={{ left: '16px' }}
+      >
+        <FaChevronLeft className="text-white" style={{ fontSize: '40px' }} />
+      </button>
+
+      <div
+        className="flex transition-transform duration-500 ease-in-out"
+        style={{
+          transform: `translateX(-${currentSlide * (100 / images.length)}%)`,
+          width: `${images.length * 100}%`,
+        }}
+      >
+        {images.map((image: string, index: number) => (
+          <div
+            className="relative w-full flex-shrink-0"
+            key={index}
+            style={{ flex: '0 0 auto', width: `${100 / images.length}%` }}
+          >
+            <Image
+              alt={`slide ${index + 1}`}
+              height={1080}
+              loading={index === 0 ? 'eager' : 'lazy'}
+              width={1920}
+              src={image}
+            />
+          </div>
+        ))}
+      </div>
+
+      <button
+        className={`next absolute top-1/2 -translate-y-1/2 transform ${
+          hovered ? 'opacity-100' : 'opacity-0'
```

Comment:
Read up on the [group Tailwind class](https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-parent-state) which can be used instead of maintaining a hover state just for adding these classes.